src/main/java/frc/robot/subsystems/Drive.java

```diff
@@ -328,23 +401,85 @@ public void setOpenloopRamp(double speed) {
     mRightMaster.configOpenloopRamp(speed);
   }
 
+  /**
+   * WPILib's arcade drive. We need this for auto turning because it allows us to set a rotation
+   * speed. Note that the deadband functionality has been removed, since we don't have to worry
+   * about driver error during auto. If we were using WPILib's {@link
+   * edu.wpi.first.wpilibj.drive.DifferentialDrive DifferentialDrive} we wouldn't need to copy this
+   * over. The output is also clamped so that we don't lose control.
+   *
+   * @param xSpeed The robot's speed along the X axis [-1.0..1.0]. Forward is positive.
+   * @param zRotation The robot's rotation rate around the Z axis [-1.0..1.0]. Clockwise is
+   *     positive.
+   * @param squareInputs If set, decreases the input sensitivity at low speeds.
+   */
+  public void arcadeHack(double xSpeed, double zRotation, boolean squareInputs) {
+    xSpeed = MathUtil.clamp(xSpeed, -1.0, 1.0);
+
+    zRotation = MathUtil.clamp(zRotation, -1.0, 1.0);
+
+    // Square the inputs (while preserving the sign) to increase fine control
+    // while permitting full power.
+    if (squareInputs) {
+      xSpeed = Math.copySign(xSpeed * xSpeed, xSpeed);
+      zRotation = Math.copySign(zRotation * zRotation, zRotation);
+    }
+
+    double leftMotorOutput;
+    double rightMotorOutput;
+
+    double maxInput = Math.copySign(Math.max(Math.abs(xSpeed), Math.abs(zRotation)), xSpeed);
+
+    if (xSpeed >= 0.0) {
+      // First quadrant, else second quadrant
+      if (zRotation >= 0.0) {
+        leftMotorOutput = maxInput;
+        rightMotorOutput = xSpeed - zRotation;
+      } else {
+        leftMotorOutput = xSpeed + zRotation;
+        rightMotorOutput = maxInput;
+      }
+    } else {
+      // Third quadrant, else fourth quadrant
+      if (zRotation >= 0.0) {
+        leftMotorOutput = xSpeed + zRotation;
+        rightMotorOutput = maxInput;
+      } else {
+        leftMotorOutput = maxInput;
+        rightMotorOutput = xSpeed - zRotation;
+      }
+    }
+
+    final double max = 0.7;
+
+    setSimplePercentOutput(
+        new DriveSignal(
+            MathUtil.clamp(leftMotorOutput, -max, max),
+            MathUtil.clamp(rightMotorOutput, -max, max)));
+  }
+
   /*
       Test all Sensors in the Subsystem
   */
   public void checkSubsystem() {}
 
-  public synchronized double getLeftEncoderDistance() {
+  public synchronized int getLeftEncoderDistance() {
     return mLeftMaster.getSelectedSensorPosition();
   }
 
-  public synchronized double getRightEncoderDistance() {
+  public synchronized int getRightEncoderDistance() {
     return mRightMaster.getSelectedSensorPosition();
   }
 
   public synchronized Rotation2d getRotation() {
     return null;
   }
 
+  public void zeroEncoders() {
+    mLeftMaster.getSensorCollection().setQuadraturePosition(0, 250);
+    mRightMaster.getSensorCollection().setQuadraturePosition(0, 250);
```

Comment:
What's with the 250 magic number? Convert to a constant to make it more descriptive.